android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt

```diff
@@ -780,6 +780,16 @@ abstract class EngineSession(
      */
     abstract fun toggleDesktopMode(enable: Boolean, reload: Boolean = false)
 
+    /**
+     * Checks if there is a rule for handling a cookie banner for the current website in the session.
+     *
+     * @param onSuccess callback invoked if the engine API returned a valid response. Please note
+     * that the response can be null - which can indicate a bug, a miscommunication
+     * or other unexpected failure.
+     * @param onError callback invoked if there was an error getting the response.
+     */
+    abstract fun hasCookieBannerRuleForSession(onSuccess: (Boolean?) -> Unit, onError: (Throwable) -> Unit)
```

Comment:
Drive-by 🚗 : we use `onSuccess` and `onError` as an existing pattern, however here, since the result is a boolean, this API looks a bit confusing.

It reads a bit like, "the result of hasCookieBannerRuleForSession if true will return in `onSuccess` and false in `onError`". Maybe I'm reading too much into it, and in which case, ignore. 🙂 

On the other hand, possibly consider two options for naming:
1. Rename `onError` to `onException` so it's clear the result from the second callback is an exceptional one.
2. Rename `onSuccess` to `onResult` so it's then clear where to expect the result of the boolean to come from (preferable).

Reply:
Thanks ❤️ , completely agree!
`onResult` reads much nicer!